projects/vdk-plugins/vdk-ipython/src/vdk_ipython.py

```diff
@@ -7,28 +7,55 @@
 from IPython.core.magic_arguments import argument
 from IPython.core.magic_arguments import magic_arguments
 from IPython.core.magic_arguments import parse_argstring
-from vdk.internal.builtin_plugins.run.standalone_data_job import (
-    StandaloneDataJobFactory,
-)
+from vdk.internal.builtin_plugins.run import standalone_data_job
+
+
+class JobControl:
+    """
+    The main idea of thi class is to make it easier to  work with IStandaloneDataJob instances
+    Since the user will be using this class
+    """
+
+    def __init__(
+        self,
+        path: str = None,
+        name: str = None,
+        arguments: str = None,
+        template: str = None,
+    ):
+        path = pathlib.Path(path) if path else pathlib.Path(os.getcwd())
+        self.job = standalone_data_job.StandaloneDataJob(
+            name=name,
+            job_args=arguments,
+            template_name=template,
+            data_job_directory=path,
+        )
+
+    def get_initialized_job_input(self):
+        return self.job.__enter__()
+
+    def finalise(self):
```

Comment:
may consider improving the `StandaloneDataJob` class, since you need to use it, and it seems that you have found a better interface surface to do so? For example, you think `finalize()` is a better way have an `__exit__` alias?

Reply:
Well, this interface surface fits vdk_ipython use of StandaloneDataJob but I am not sure whether it is so for all use case scenarios of the class. If I remember correctly the main idea of using __enter__ and __exit__ in this class is to use it with the StadaloneDataJobFactory (which I do not use in this solution)
I don't know whether it will be ok to change the whole class for this specific interface surface
We might get more opinion, @tozka what do yo. think about it?

Reply:
It's an adapter. It infers things like job directory , name etc from the notebook location. 

I am tno sure about the names. JobControl is internal implementation name and won't be exposed to end user so it's less important what is it. 

The end user would use `job. get_initialized_job_input()` and `job.finalize()`

We probably would need to iterate on the names and looks for user feedback .